taxonomy-editor-frontend/src/pages/editentry/ListTranslations.jsx

```diff
@@ -1,94 +1,351 @@
+    const [mainLang_renderedTranslations, setMainLang_renderedTranslations] = useState([]) // Stores state of main language's tags
+    const [open, setOpen] = useState(false); // Used for Dialog component
+    const [newLanguageCode, setNewLanguageCode] = useState(''); // Used for storing new LC from Dialog
+    const [isValidLanguageCode, setisValidLanguageCode] = useState(false); // Used for validating a new LC
+    const [btnDisabled, setBtnDisabled] = useState(true) // For enabling or disabling Dialog button
+
+    // Helper functions for Dialog component
+    function handleClose() { setOpen(false); }
+    function handleOpen() { setOpen(true); }
+
+    // Used for addition of a translation language
+    function handleAddTranslation(key) {
+        const newRenderedTranslations = [...renderedTranslations, {'languageCode' : key, 'tags' : []}]
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
         
-        // Get all tags and its corresponding language code
-        // Tagids need to be recomputed, so shouldn't be rendered
-        // Main language isn't considered, since it's rendered separately
- 
-        if (key.startsWith('tags') && 
-            !key.endsWith(nodeObject.main_language) && 
-            !key.includes('ids')) {
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            newNodeObject[key] = [];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Used for deleting a translation language
+    function handleDeleteTranslation(key) {
+        const newRenderedTranslations = renderedTranslations.filter(obj => !(key === obj.languageCode))
+        setRenderedTranslations(newRenderedTranslations);
+        key = 'tags_' + key; // LC must have a prefix "tags_"
+        
+        // Make changes to the parent NodeObject
+        setNodeObject(prevState => {
+            const newNodeObject = {...prevState};
+            delete newNodeObject[key];
+            return newNodeObject
+        })
+        setOpen(false);
+    }
+
+    // Changes the translations to be rendered
+    // Dependent on changes occuring in "originalNodeObject"
+    useEffect(() => {
+
+        // Main langauge tags are considered separately, since they have to be rendered first
+        let mainLangTags = []
+        let otherLangTags = []
+
+        Object.keys(originalNodeObject).forEach((key) => {
+        
+            // Get all tags and its corresponding language code
+            // Tagids need to be recomputed, so shouldn't be rendered
+            // Eg: tags_fr
+        
+            if (key.startsWith('tags') && !key.includes('ids')) {
+
+                // If tags are not for the main language, add them to otherLangTags
+                if (!key.endsWith(originalNodeObject.main_language)) {
```

Comment:
nit: `if(key.includes(originalNodeObject.main_language)) {...// main language} else { //not main language}`  it is usually easier to follow the `if (contidion === true) {}` than `if (negate condition) {} else { // which would be the condition === true anyways}`